sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCallGCSuppressFinalize.NetCore.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    // https://github.com/SonarSource/sonar-dotnet/issues/3639
+    public class Custom : IAsyncDisposable
+    {
+        public void Method()
+        {
+            GC.SuppressFinalize(this); // Noncompliant
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore();
+
+            GC.SuppressFinalize(this); // Noncompliant - FP, see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method
+        }
+
+        protected virtual ValueTask DisposeAsyncCore() => default;
+    }
+
+    public class Fake
+    {
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore();
```

Comment:
This line causes compilation error for the snippet. Can it be removed? I don't think it's needed for purpose of this test case. 
```suggestion
```

Reply:
It is there to avoid warnings for the `async` keyword but I forgot to add `DisposeAsyncCore` method